src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -410,30 +406,35 @@ private void BuildLinkTag(TagHelperAttributeList attributes, TagHelperContent bu
             for (var i = 0; i < attributes.Count; i++)
             {
                 var attribute = attributes[i];
-                var attributeValue = attribute.Value;
+
+                // Ignore the href in the existing attributes dictionary, it's handled below.
                 if (AppendVersion == true &&
                     string.Equals(attribute.Name, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
                 {
-                    // "href" values come from bound attributes and globbing. So anything but a non-null string is
-                    // unexpected but could happen if another helper targeting the same element does something odd.
-                    // Pass through existing value in that case.
-                    var attributeStringValue = attributeValue as string;
-                    if (attributeStringValue != null)
-                    {
-                        attributeValue = _fileVersionProvider.AddFileVersionToPath(attributeStringValue);
-                    }
+                    continue;
```

Comment:
Why is it :ok: to reorder the attributes in the `LinkTagHelper` but not in the `ScriptTagHelper`? Prefer choice made in the `ScriptTagHelper`.